numpy/lib/npyio.py

```diff
@@ -861,16 +876,20 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
     # Type conversions for Py3 convenience
     if comments is not None:
         if isinstance(comments, (basestring, bytes)):
-            comments = [asbytes(comments)]
-        else:
-            comments = [asbytes(comment) for comment in comments]
+            comments = [comments]
+
+        comments = [_decode_line(x) for x in comments]
```

Comment:
Should this forward the `encoding` argument? (after the handling of `"bytes"` below)

Reply:
I don't think so. The `encoding` parameter refers to the file encoding, while the comment string is passed as an argument. A better option might be to not accept bytes strings for this.

Reply:
I'll update the docstring to mention the default encoding. The previously used `asbytes` assumed 'latin1'.

Reply:
Hmm, maybe not. There are a bunch of other passed string parameters, none of which are explicitly decoded if they are byte strings. The only thing special about `comments` is that it may be a sequence. I think we should simply drop the byte string option.

Reply:
OK, what I have done is decoded the `delimiter` also and added a note to the documentation that if either of `delimiter` or `comments` are passed as byte strings, that they will be decoded as 'latin1'.  We might run into some backwards compatibility problems here, but I don't expect many folks were passing byte strings, especially as that would assume that the input file has some strange encoding that couldn't be handled by the `latin1` assumption. We should maybe warn if either is a byte string.